drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -121,6 +132,30 @@ local function temperature_attr_handler(driver, device, ib, response)
   end
 end
 
+local temp_attr_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value == nil then
+      return
+    end
+    local temp = ib.data.value / 100.0
+    local unit = "C"
+    set_field_for_endpoint(device, TEMP_BOUND_RECEIVED..minOrMax, ib.endpoint_id, temp)
+    local min = get_field_for_endpoint(device, TEMP_BOUND_RECEIVED..TEMP_MIN, ib.endpoint_id)
+    local max = get_field_for_endpoint(device, TEMP_BOUND_RECEIVED..TEMP_MAX, ib.endpoint_id)
+    -- Only emit the capability for RPC version >= 5 (unit conversion for
+    -- temperature range capability is only supported for RPC >= 5)
+    if version.rpc >= 5 and min ~= nil and max ~= nil then
+      if min < max then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureMeasurement.temperatureRange({ value = { minimum = min, maximum = max }, unit = unit }))
+        set_field_for_endpoint(device, TEMP_BOUND_RECEIVED..TEMP_MIN, ib.endpoint_id, nil)
```

Comment:
These values are used in `set_setpoint`, so I think this `nil` setting needs to be removed